lib/matplotlib/tests/test_axes.py

```diff
@@ -1472,6 +1473,34 @@ def test_bar_tick_label_multiple_old_alignment():
            align='center')
 
 
+@check_figures_equal(extensions=["png"])
+def test_bar_decimal_center(fig_test, fig_ref):
+    ax = fig_test.subplots()
+    x0 = [1.5, 8.4, 5.3, 4.2]
+    y0 = [1.1, 2.2, 3.3, 4.4]
+    x = [Decimal(x) for x in x0]
+    y = [Decimal(y) for y in y0]
+    # Test image - vertical, align-center bar chart with Decimal() input
+    ax.bar(x, y, align='center')
+    # Reference image
+    ax = fig_ref.subplots()
+    ax.bar(x0, y0, align='center')
+
+
+@check_figures_equal(extensions=["png"])
```

Comment:
I don't think testing both bar and barh is particularly necessary and would just keep one of them.

Reply:
I this there will be no problem since width will also be converted. In the following code, if width is of type Decimal, it will be converted to floats by convert_xunits. 
```
 if self.xaxis is not None:
    x0 = x
    x = np.asarray(self.convert_xunits(x))
    width = self._convert_dx(width, x0, x, self.convert_xunits)
    ......
```
Similarly for height.

Reply:
I can generate plots with float x/y but Decimal width/height as this:
![image](https://user-images.githubusercontent.com/21251857/57577261-34db6300-7441-11e9-8367-fb0b04aec426.png)